### PR TITLE
Kp/contributing consolidation in dev guide

### PR DESCRIPTION
**Pull request by PipKat**
Made a few changes, primarily to ensure that I had write permission to the repo.

### CODE REVIEW

**Comment by PipKat** [comment]
@Alex--I was able to submit this PR for your review.

File: doc/source/guidelines/dev_practices.rst [hunk 1]
```diff
@@ -2,102 +2,243 @@
...
+#. Test your contribution. Because Python is an interperted language, if
+   it's not tested, it's probably broken. At the minimum, include a unit
+   test for each new feature within the ``tests`` directory. Ensure that
+   each new method, class, or function has reasonable (>80%) coverage.
+   See `Testing <#Testing>`__ for automating testing.
+
+#. Do not include any datasets for which a license is not available
+   or commercial use is prohibited.
+
+#. Review our our `Code of Conduct <https://github.com/pyansys/DPF-Core/blob/master/CODE_OF_CONDUCT.md>`_.
```
Comment (MaxJPRey) [inline doc/source/guidelines/dev_practices.rst:30]:
```suggestion
#. Review our `Code of Conduct <https://github.com/pyansys/DPF-Core/blob/master/CODE_OF_CONDUCT.md>`_.
```

File: doc/source/guidelines/dev_practices.rst [hunk 2]
```diff
@@ -2,102 +2,243 @@
...
+#. Follow the `Zen of Python <https://www.python.org/dev/peps/pep-0020/>`__.
+   As silly as core Python developers are sometimes, there's much to be
+   gained by following the basic guidelines listed in PEP 20. As suggested
+   in these guidelines, focus on making your additions intuitive, novel,
+   and helpful for PyAnsys library users. When in doubt, use ``import this``.
+   For Ansys code quality standards, see :ref:`coding_style`.
+
+#. Document your contributions. Include a docstring for any added
+   function, method, or class, following `numpydocs docstring <https://numpydoc.readthedocs.io/en/latest/format.html>`_
+   guidelines and PyAnsys documenation standards <#Documentation Standards>.
```
Comment (MaxJPRey) [inline doc/source/guidelines/dev_practices.rst:18]:
```suggestion
   guidelines and PyAnsys documentation standards <#Documentation Standards>.
```

File: doc/source/guidelines/dev_practices.rst [hunk 3]
```diff
@@ -2,102 +2,243 @@
...
+via GitHub Actions.
+
+Documentation for the latest stable release of a PyAnsys libary is accessible
+from its repository. You can access the latest development version of the
+documentation tracking the ``main`` branch by adding the prefix ``dev.`` to
+the URL for the latest stable release. For example, the URL for the latest
+stable release of the PyAEDT documentation is
+''https://dev.aedtdocs.pyansys.com/``, and the URL for the latest development 
+version of this documenation is ''https://dev.aedtdocs.pyansys.com/``. The
+latest develompment version is kept up-to-date automatically via GitHub actions.
```
Comment (MaxJPRey) [inline doc/source/guidelines/dev_practices.rst:169]:
```suggestion
latest development version is kept up-to-date automatically via GitHub actions.
```

File: doc/source/guidelines/dev_practices.rst [hunk 4]
```diff
@@ -2,102 +2,243 @@
...
+examples run properly locally as they will be verified through the CI performed
+via GitHub Actions.
+
+Documentation for the latest stable release of a PyAnsys libary is accessible
+from its repository. You can access the latest development version of the
+documentation tracking the ``main`` branch by adding the prefix ``dev.`` to
+the URL for the latest stable release. For example, the URL for the latest
+stable release of the PyAEDT documentation is
+''https://dev.aedtdocs.pyansys.com/``, and the URL for the latest development 
+version of this documenation is ''https://dev.aedtdocs.pyansys.com/``. The
```
Comment (MaxJPRey) [inline doc/source/guidelines/dev_practices.rst:168]:
```suggestion
version of this documentation is ''https://dev.aedtdocs.pyansys.com/``. The
```

File: doc/source/guidelines/dev_practices.rst [hunk 5]
```diff
@@ -109,19 +250,20 @@ Descriptions follow for the main features of the branching model.
...
-#. Locally run all tests as outlined in :ref:`testing` and ensure all
-   are passing.
+#. Locally run all tests as outlined in `Testing <#Testing>`_ and
+   ensure that all are passing.
 
-#. Locally test and build the documentation with link checking to make
-   sure no links are outdated. Be sure to run ``make clean`` to ensure no
-   results are cached.
+#. Locally test and build the documentation with link checking to
+   ensure that no links are outdated
```
Comment (MaxJPRey) [inline doc/source/guidelines/dev_practices.rst:264]:
```suggestion
   ensure that no links are outdated.
```

File: doc/source/guidelines/dev_practices.rst [hunk 6]
```diff
@@ -133,36 +275,36 @@ Procedures follow for major and minor releases.
...
+	     git tag v<MAJOR.MINOR.0>
         git push origin --tags
 
 #. Create a list of all changes for the release. It is often helpful
-   to see the differences from the last tag and the ``main`` branch
-   using `GitHub's compare feature`_ tool.  Be sure to acknowledge new
-   contributors by their GitHub username and place mentions where
-   appropriate if a specific contributor is to be thanked for a new
-   feature.
+   to levearage GitHub's compare feature to see the differences from
```
Comment (MaxJPRey) [inline doc/source/guidelines/dev_practices.rst:300]:
```suggestion
   to leverage GitHub's compare feature to see the differences from
```

File: doc/source/overview/contributing.rst [hunk 7]
```diff
@@ -2,111 +2,123 @@
...
+our libraries powerful tools for all users. As such, we
+encourage you to submit questions, report bugs, request new
+features, contribute code, and start discussions.
+
+This page provides general information about contributing to a
+PyAnsys library. Contribution information specific to a particular
+library appears on the ``Contributing`` page in the respective
+library's documentation, including:
+
+- Instrutions for cloning the source repository from GitHub
```
Comment (MaxJPRey) [inline doc/source/overview/contributing.rst:18]:
```suggestion
- Instructions for cloning the source repository from GitHub
```

File: doc/source/overview/contributing.rst [hunk 8]
```diff
@@ -2,111 +2,123 @@
...
-code to this repository and all PyAnsys library repositories.
+- `PyAEDT Issues <https://github.com/pyansys/pyaedt/issues>`_
+- `PyDPF-Core Issues <https://github.com/pyansys/pydpf-core/issues>`_
+- `PyDPF-Post Issues <https://github.com/pyansys/pydpf-post/issues>`_
+- `PyMAPDL Issues <https://github.com/pyansys/pymapdl/issues>`_
+- `PyMAPDL Reader Issues <https://github.com/pyansys/pymapdl-reader/issues>`_
 
+The overall PyAnsys project support team can be reached at
+pyansys.support@ansys.com. Someone will either respond or direct the
+message to the respective libary's support team.
```
Comment (MaxJPRey) [inline doc/source/overview/contributing.rst:33]:
```suggestion
message to the respective library's support team.
```

File: doc/source/overview/contributing.rst [hunk 9]
```diff
@@ -2,111 +2,123 @@
...
+- :ref:`best_practices` for information on how to style and format your
+  code to adhere to PyAnsys standards
+
+Starting Discussions
+--------------------
+In the repository for the respective PyAnsys library, you should raise
+general questions regarding development practices as discussions rather
+than issues. For example, raise general questions about PyMAPDL
+in `PyMAPDL Discussions <https://github.com/pyansys/pymapdl/discussions>`_. 
+Dedpending on what is decided, issues can be spun out of general discussions.
```
Comment (MaxJPRey) [inline doc/source/overview/contributing.rst:84]:
```suggestion
Depending on what is decided, issues can be spun out of general discussions.
```

**Review by MaxJPRey — APPROVED** [review]

**Comment by MaxJPRey** [comment]
LGTM.

File: doc/source/guidelines/dev_practices.rst (hunk 4 above)
Comment (MaxJPRey) [inline doc/source/guidelines/dev_practices.rst:168]:
```suggestion
`<https://dev.aedtdocs.pyansys.com/>`_, and the URL for the latest development 
version of this documenation is `<https://dev.aedtdocs.pyansys.com/>`_. The
```

**Comment by PipKat** [comment]
@MaxJPRey Thanks for catching so many typos! I was working late on this Friday and obviously wasn't careful. Wonder why the spellcheck didn't catch a lot of these?

**Pull request merged**